### PR TITLE
add support for SOAP binding

### PR DESCRIPTION
**Pull request by annashamray**
changes:
* add `Soap_Logout_Request` which process request message from Idp with SOAP binding
* support 2 different SLO endpoints for sp metadata

### CODE REVIEW

File: src/onelogin/saml2/xml_utils.py [hunk 1]
```diff
@@ -175,3 +176,40 @@ def element_text(node):
...
+        """
+
+        soap_xml = OneLogin_Saml2_XML.to_etree(xml)
+
+        return OneLogin_Saml2_XML.query(soap_xml, "/soap:Envelope/soap:Body")[
+            0
+        ].getchildren()[0]
+
+    @staticmethod
+    def add_soap_envelop(xml: str) -> str:
```
Comment (SilviaAmAm) [inline src/onelogin/saml2/xml_utils.py:195]:
```suggestion
    def add_soap_envelope(xml: str) -> str:
```

File: src/onelogin/saml2/xml_utils.py [hunk 2]
```diff
@@ -175,3 +176,40 @@ def element_text(node):
...
+        """
+        wraps xml in the SOAP Envelop
+
+        :param xml: is xml which should be put in SOAP Envelop
```
Comment (SilviaAmAm) [inline src/onelogin/saml2/xml_utils.py:199]:
```suggestion
        wraps xml in the SOAP Envelope

        :param xml: is xml which should be put in SOAP Envelope
```

**Review by SilviaAmAm — APPROVED** [review]
Very nice :slightly_smiling_face:

File: src/onelogin/saml2/metadata.py [hunk 3]
```diff
@@ -163,12 +163,22 @@ def builder(cls, sp, authnsign=False, wsign=False, valid_until=None, cache_durat
...
+                    {
+                        'binding': sp['singleLogoutService']['binding'],
+                        'location': sp['singleLogoutService']['url'],
+                    }
+                sls += sls_logout_request
+
+            if 'responseUrl' in sp['singleLogoutService']:
+                sls_logout_response = OneLogin_Saml2_Templates.MD_SLS % \
+                     {
+                         'binding': sp['singleLogoutService'].get("responseBinding", "Binding"),
```
[the remark this answers is not in the document]
Comment (annashamray) [reply src/onelogin/saml2/metadata.py:178]: Hmm you are right it doesn't make sense. I don't remember why I did it. I'll remove the default

**Review by SilviaAmAm — APPROVED** [review]

**Pull request merged**